zml/platform.zig

```diff
@@ -84,3 +86,78 @@ pub const Platform = struct {
         return self.pjrt_client.getProfiler(self.pjrt_api, options);
     }
 };
+
+const _CreateOptions = struct {
+    // XLA CPU client doesn't read options
+    // https://github.com/openxla/xla/blob/42496a28c374bd35f493cc5dbde74805407245dc/xla/pjrt/c/pjrt_c_api_cpu_internal.cc#L33-L46
+    cpu: NoOpt = .{},
+
+    // match XLA defaults
+    // https://github.com/openxla/xla/blob/3e87afa11a865cf91137522492918ad18bfe5b7c/xla/pjrt/plugin/xla_gpu/xla_gpu_allocator_config.h#L25-L60
+    cuda: Cuda = .{ .allocator = .{ .bfc = .{ .preallocate = true, .memory_fraction = 0.75 } } },
+    rocm: NoOpt = .{},
+    tpu: NoOpt = .{},
+    neuron: NoOpt = .{},
+
+    // NoOpt instead of void to allow it to work with json. Yes, it's a bit annoying.
+    pub const NoOpt = struct {};
+
+    pub const Cuda = struct {
+        // Use the same default than XLA.
+        allocator: Allocator = .{ .bfc = .{} },
+        // TODO support all of https://github.com/openxla/xla/blob/3d31c48c719d331d432132b3e0c2c5ce52650675/xla/pjrt/c/pjrt_c_api_gpu_internal.cc#L76-L86
+        visible_devices: []const u16 = &.{},
+        // node_id
+        // num_nodes
```

Comment:
@AdjectiveAllison suggested adding at least this ^^^ `num_nodes`

Reply:
I agree with the feeling,
I've started https://github.com/zml/zml/pull/132 but it's not working as expected

in the interest of moving forward, I'll merge this PR and follow up later